proxysql/test-proxysql

```diff
@@ -38,45 +38,66 @@ function process_proxysql_output {
+
+# Installing PXC-5.7 client
+  sudo wget https://repo.percona.com/apt/percona-release_latest.generic_all.deb
+  sudo dpkg -i percona-release_latest.generic_all.deb
+  sudo percona-release enable original
+  sudo apt-get update -y
+  sudo apt-get install -y percona-xtradb-cluster-57
 fi
 
-tar -C ${WORKDIR_ABS}/WORKDIR/PROXY  --strip-components=1 -zxpf $(ls $WORKDIR_ABS/proxysql*.tar.gz | head -1)
+if [ -f /usr/bin/yum ]; then
+  sudo yum remove -y Percona-Server-shared-56-5.6.49-rel89.0.1.el7.x86_64
+  if [[ $RHEL -eq 8 ]]; then
+    sudo yum remove -y mariadb-connector-c-config-3.0.7-1.el8.noarch
+  fi
+  # TODO: For CentOS machine support both PXC5.7 and 8.0
+    sudo yum install -y percona-xtrabackup-24 Percona-XtraDB-Cluster-57 proxysql2
+else
+# Installing proxysql package
+  sudo apt update -y
+  sudo wget -q -O ${WORKDIR_ABS}/proxysql_2.0.17-ubuntu20_amd64.deb https://repo.proxysql.com/ProxySQL/proxysql-2.0.x/focal/proxysql_2.0.17-ubuntu20_amd64.deb
+  sudo apt-get remove proxysql2
+  sudo apt-get purge proxysql2
+  sudo dpkg -i ${WORKDIR_ABS}/proxysql_2.0.17-ubuntu20_amd64.deb
+fi
 
 pushd ${WORKDIR_ABS}/WORKDIR
-sudo tar -xzf $WORKDIR_ABS/proxysql*.tar.gz -C ${WORKDIR_ABS}/WORKDIR
-sudo cp $WORKDIR_ABS/WORKDIR/PROXY/etc/proxysql.cnf /etc/
-sudo cp $WORKDIR_ABS/WORKDIR/PROXY/etc/proxysql-admin.cnf /etc/
-sudo cp $WORKDIR_ABS/WORKDIR/PROXY/usr/bin/* /usr/bin
-cp $WORKDIR_ABS/WORKDIR/PROXY/usr/bin/proxysql-* $WORKDIR_ABS/WORKDIR/
-sudo chmod 644 /etc/proxysql*.cnf
+sudo tar -xzf $WORKDIR_ABS/proxysql*.tar.gz -C ${WORKDIR_ABS}/PROXY_ADMIN
+sudo ln -s $WORKDIR_ABS/PROXY_ADMIN/proxysql*/usr/bin/proxysql-admin ${WORKDIR_ABS}/WORKDIR/proxysql-admin
+sudo ln -s $WORKDIR_ABS/PROXY_ADMIN/proxysql*/usr/bin/proxysql-admin-common ${WORKDIR_ABS}/WORKDIR/proxysql-admin-common
+sudo ln -s $WORKDIR_ABS/PROXY_ADMIN/proxysql*/usr/bin/proxysql-login-file ${WORKDIR_ABS}/WORKDIR/proxysql-login-file
+sudo ln -s $WORKDIR_ABS/PROXY_ADMIN/proxysql*/etc/proxysql-admin.cnf ${WORKDIR_ABS}/WORKDIR/proxysql-2.0/etc/proxysql-admin.cnf
+sudo ln -s /usr/bin/proxysql $WORKDIR_ABS/WORKDIR/proxysql-2.0/usr/bin/proxysql
 status=0
+
 # Running ProxySQL QA script
-if [ -f /usr/bin/yum ]; then
-    sudo cp /etc/hosts ${WORKDIR_ABS}/hosts.new
-    sudo sed -i 's/::1*.*localhost /::1 /' ${WORKDIR_ABS}/hosts.new
-    sudo cp -f ${WORKDIR_ABS}/hosts.new /etc/hosts
-fi
-sudo sed -i 's/wsrep_sst_method=rsync/wsrep_sst_method=xtrabackup-v2/g' $WORKDIR_ABS/WORKDIR/PROXY/tests/proxysql-admin-testsuite.sh
-sudo bash $WORKDIR_ABS/WORKDIR/PROXY/tests/proxysql-admin-testsuite.sh \
+sudo cp /etc/hosts ${WORKDIR_ABS}/hosts.new
+sudo sed -i 's/::1*.*localhost /::1 /' ${WORKDIR_ABS}/hosts.new
+sudo cp -f ${WORKDIR_ABS}/hosts.new /etc/hosts
```

Comment:
why do we need these 2 cp commands?

Reply:
Or else it complains about resouce being busy
+ sudo sed -i 's/::1*.*localhost /::1 /' /etc/hosts
sed: cannot rename /etc/sed2KyKQ3: Device or resource busy

Reply:
Because the /etc/hosts file is bind-mounted to host's one
It can't be directly changed inside container.

Reply:
ok